### DISCUSSION TITLE
Adopt conventional commits and semantic releases?

### DISCUSSION THREAD

**Discussion by josias-r** [directus/directus#3572] [Feature Requests]
I absolutely love this project as well as the work that is put into it 😁

I was wondering if you (the team behind it) would consider adopting conventional commits and semantic releases.
Especially now, during the RC releases stage, some automation and standard like that would be great to have.

Though I fully understand that this might not line up with your management and maybe it is not a priority you guys currently have, I would love to contribute in case you are not against this idea.

The main reason I'm asking because right now, the RC releases, for instance, do not have any changelogs. This could be automated by working with conventional commits. Also, from how I see it, it seems to be the most widespread standard to work with semantic releases (that + conventional commits go hand in hand). For example, instead of creating `rc` releases, you would create `@next` releases (and install them using `npm i directus@next`)

Just as an example if you haven't heard of this, it basically works like this:

1. Commit messages should follow [a strict standard](https://www.conventionalcommits.org/en/v1.0.0/)
2. Based on the commit messages the changelog will get generated + the version number is automatically incremented
3. Releases are created by pipelines/actions that run on push
4. Based on the branch you push to, the automation tool knows what kind of release it is (for example push to the `next` branch to create a `@next` release, or push to the master to create a final release)

**Comment by benhaynes** [comment]
Thanks for this! We're always trying to improve how we work, and add good standards to our platform/workflow. Though, being on the more technical side, I defer to @rijkvanzanten on this one...

**Comment by rijkvanzanten** [comment]
We actually used to do this back in the v7 days, and it didn't work for us in practice. More often than not, external contributors would ignore / forget about the commit rules, and have (very valuable) PRs blocked just because of that.. Secondly, in a project the size of Directus, there's multiple "gradations" of "breaking-ness" in changes. I know that sounds weird, but we have to consider where and what a change affects in order to confidently say it's a breaking change for the community. (For example, a type change in a parameter for an extension is theoretically a breaking change, but doesn't warrant a full major version bump, as it doesn't necessarily affect the user base). That makes it really difficult to rely on a "black and white" automated way of controlling what the next version should be.

As for changelogs, we aim to have a milestone for each release (see https://github.com/directus/directus/milestones?state=closed), which also acts as a changelog for older versions.

**Reply by josias-r** [reply]
The problem with contributors can be solved by working with squash commits and creating clean commits as the admin. Though I can see that in some cases this could lead to additional work for you guys and therefore not be ideal.

But your explanation makes perfect sense and I guess this would now be a really subjective topic.

My main reason, as I mentioned, was the "missing" changelog for this idea/proposal. But since you work closely with milestones, I can find the changes with the link you provided (maybe add a changelog.md linking to the milestones or mention it with the release messages?). Thank you for the info! 😃

**Reply by rijkvanzanten** [reply]
Yeah! Might be nice to look into setting up a GH action that automatically pulls in the issues from the corresponding milestone and plops them into the release 🤔

**Reply by josias-r** [reply]
Something like that would be awesome :D

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
